src/api/mod.rs

```diff
@@ -377,13 +380,23 @@ pub trait Peripheral: Send + Sync + Clone + Debug {
 }
 
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-#[derive(Debug, Copy, Clone)]
```

Comment:
I removed the `Copy` trait from here so that a `Vec<u8> may be supplied to the *Advertisement variants. `Vec` does not support shallow copying